### ISSUE TITLE
[BUG] npx with github scheme will fail without any message

### ISSUE THREAD

**Issue by anatawa12** [npm/cli#5749]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### This issue exists in the latest npm version

- [X] I am using the latest npm

### Current Behavior

```sh
npx github:viperproject/check-license-header#v1
```

falls (exits 1) without any error message

### Expected Behavior

```sh
npx github:viperproject/check-license-header#v1
```

run [viperproject/check-license-header](https://github.com/viperproject/check-license-header)

### Steps To Reproduce

1. With node 16.18.0 and npm 8.19.2
2. With no (additional) configuration
3. Run `npx github:viperproject/check-license-header#v1`

### Environment

- npm: 8.19.2
- Node.js: v16.18.0
- OS Name: macOS Ventura 13.0
- System Model Name: MacBookPro 14-inch, 2021
- npm config:
```ini
; node bin location = /opt/n/bin/node
; node version = v16.18.0
; npm local prefix = /Users/anatawa12/IdeaProjects/fixRTM
; npm version = 8.19.2
; cwd = /Users/anatawa12/IdeaProjects/fixRTM
; HOME = /Users/anatawa12
; Run `npm config ls -l` to show all defaults.
```

**Comment by anatawa12** [comment]
I found `npmBin` passed to `GitFetcher` in `pacote` is path to npx-cli.js, not a npm and `GitFetcher[_prepareDir]` executes `node npx-cli.js install <options....>` so `git dep preparation` will be failed.

**Comment by anatawa12** [comment]
https://github.com/npm/cli/blob/292156c60bc16ba4da5942df4dd54b1c4cf9ad72/lib/utils/config/flatten.js#L23

I think this line is the problem. the XXX says `using npm programmatically is a bit of a pain.`

**Comment by anatawa12** [comment]
And on the latest branch, here's the problem.
I've tested npm v9 but this problem still exiets.

https://github.com/npm/cli/blob/bc5ec0517eed36a9fa310c8be9d0d089afc6925e/lib/utils/config/index.js#L22-L26

**Comment by nlf** [comment]
good catch! the mix up between the npx bin and the npm bin is definitely what's broken here

**Comment by milaninfy** [comment]
This appears to be fixed on latest npm 
```sh
~ $ npx github:viperproject/check-license-header#v1 check
check-license-header check

Check license headers

Options:
      --version  Show version number                                   [boolean]
  -c, --config   Path to JSON config file                    [string] [required]
      --path     Path to working directory[string] [default: "/Users/milaninfy"]
  -h, --help     Show help                                             [boolean]
      --strict   Specifies whether files not covered by the configuration should
                 be treated as errors                 [boolean] [default: false]

Missing required argument: c
Please specify path to config file
~ $ npm -v
10.8.3
~ $ npx -v
10.8.3
```

**Comment by anatawa12** [comment]
it looks fixed by #6762
